DDD-8.md

```diff
@@ -126,3 +126,146 @@ So if the snapshot is reading a chunk from a table that is modified through a su
+
+### MongoDB
+
+Also for MongoDB the motivation behind the research a possible way to provide read-only incremental snapshot are the same
+with a specific addition: try to overcome the challenges related to the different kind of MongoDB topologies. 
+
+#### Topologies
+MongoDB has three topology types:
+
+* Standalone server
+* Replica set
+* Sharded cluster
+
+Since Standalone server do not have the _oplog_ and so Debezium is not able to streaming events from it, we do not consider it. Although MongoDB does not recommend running a standalone server in production.
+
+##### Replica set
+
+A [replica set (RS)](https://www.mongodb.com/docs/manual/core/replica-set-members/) in MongoDB is a group of _mongod_ processes that provide redundancy and high availability. The members of a replica set are:
+
+* Primary:
+  receives all write operations.
+* Secondaries:
+  replicate operations from the primary to maintain an identical data set. To replicate data, a secondary applies operations from the primary's oplog to its own data set in an asynchronous process. A replica set can have one or more secondaries.
+
+Every node in a RS has its own [_oplog_ collection](https://www.mongodb.com/docs/manual/core/replica-set-oplog/) located into `local.oplog.rs`.
+
+The standard way to connect to RS is using the following connection string
+
+```text
+mongodb://user:password@mongodb0.example.com:27017,mongodb1.example.com:27017,mongodb2.example.com:27017/?authSource=admin&replicaSet=myRepl
+```
+
+providing the complete or partial list of nodes that compose the RS. 
+In that case the client attempts to discover all servers in the RS, and sends operations to the primary member.
+
+By default, an application directs its [read operations](https://www.mongodb.com/docs/manual/core/read-preference/) to the primary member in a replica set (i.e. read preference mode "primary"). But, clients can specify a read preference to send read operations to secondaries.
+
+This type of configuration is common for CDC scenario where users wants to reduce the load on the primary that is already under the write load.
+
+So you can directly connect to a secondary specifying its address and setting the read preference mode accordingly. 
+
+This configuration can lead to a possible weird issues in particular type of deployments. For example, if you have that nodes of the RS has two network address, internal and external, and you configure 
+the connection string to point to a secondary using its external address, when the client connects to the RS and retrieve its information, even the intern addresses of the nodes, it then can use the internal address causing connection failures. 
+To overcome to this, you can set the [directConnection](https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.directConnection) property.
```

Comment:
More specifically the client will always use the addresses returned by the cluster description. For this reason with RS topology it is generally not a good idea to configure the RS with addresses which are not reachable form the outside. The correct approach in this case is to use hostnames and make sure that the hostname is resolved to internal address for in-cluster communication and resolve to a public address for external access.

Note that `directConenction` requires only a single host and sacrifices HA capabilities (if the node goes down the client will not switch to a different node in the RS).